### PR TITLE
Fix issues when itemLabelPath and itemValuePath are undefined

### LINKED ISSUE #717: Setting itemLabelPath to undefined should produce an error message
### Description
If you (accidentally) set `itemLabelPath` to undefined, the popup can never be opened. Even if you change the value later on, the popup will never open. No error message or exception will be printed either.

### Expected outcome
The popup is opened with either the default `toString` value or empty values. An error message is printed 

### Actual outcome
Overlay does not open

### Live Demo
https://glitch.com/edit/#!/thunder-barracuda?path=index.html:21:0

### PR DESCRIPTION
**Pull request by niiyeboah**
Fixes #717

---
This change is [Reviewable](https://reviewable.io/reviews/vaadin/vaadin-combo-box/763)

### CODE REVIEW

File: test/using-object-values.html [hunk 1]
```diff
@@ -259,6 +264,45 @@
...
+
+        const selectItemComboboxUndefinedPaths = (index) => {
+          // simulates clicking on the overlay items, but it more reliable in tests.
+          comboboxUndefinedPaths.$.overlay.dispatchEvent(new CustomEvent('selection-changed', {
+            detail: {item: comboboxUndefinedPaths.items[index]}
+          }));
+        };
+
+        beforeEach(() => {
+          error = sinon.stub(console, 'error');
```
Comment (tomivirkki) [inline test/using-object-values.html]: If you're stubbing `console.error` on `beforeEach`, I'd suggest restoring it in `afterEach`

Comment (tomivirkki) [reply test/using-object-values.html]: Or optionally do the stubbing in the test cases directly

File: src/vaadin-combo-box-dropdown-wrapper.html [hunk 2]
```diff
@@ -328,7 +328,7 @@
        * @return {String}
        */
       getItemLabel(item) {
-        let label = item ? this.get(this._itemLabelPath, item) : undefined;
+        let label = item && this._itemLabelPath ? this.get(this._itemLabelPath, item) : undefined;
```
Comment (tomivirkki) [inline src/vaadin-combo-box-dropdown-wrapper.html]: This is better than before but the intended logic still doesn't work correctly.
1. Open "Filtering with a Remote Data Source" (It uses actual objects as items)
1.1. Open the combo box once so the items are listed
2. Set `itemLabelPath` to undefined (Now this produces an error message but the labels are not updated (to the intended toString() output))
3. Set `itemLabelPath` to "link", which is a valid item subproperty -> The error goes away but the items still show old values

File: src/vaadin-combo-box-mixin.html [hunk 3]
```diff
@@ -623,10 +624,29 @@
       }
     }
 
+    _itemLabelPathChanged(itemLabelPath, oldItemLabelPath) {
```
Comment (tomivirkki) [inline src/vaadin-combo-box-mixin.html]: There should already be functionality somewhere that refreshes the item labels since they're updated for example when the items change. No need to write new logic for it but instead make sure that the existing logic gets invoked also on `itemLabelPath` change

Comment (niiyeboah) [reply src/vaadin-combo-box-mixin.html]: Done

File: src/vaadin-combo-box-mixin.html [hunk 4]
```diff
@@ -623,10 +624,21 @@
...
+        console.error('You should set itemLabelPath to a valid string');
+      }
+
+      if (this.selectedItem) {
+        this._inputElementValue = this._getItemLabel(this.selectedItem);
+      }
+    }
+
     _filterChanged(filter, itemValuePath, itemLabelPath) {
-      if (filter === undefined || itemValuePath === undefined || itemLabelPath === undefined) {
```
Comment (web-padawan) [inline src/vaadin-combo-box-mixin.html]: Why was this line changed? Especially, `itemValuelPath` change seems unrelated.
Same as above: please revert the change to this line in case if new logic does not require it.

File: src/vaadin-combo-box-mixin.html [hunk 5]
```diff
@@ -742,7 +754,7 @@
     }
 
     _itemsOrPathsChanged(e, itemValuePath, itemLabelPath) {
-      if (e.value === undefined || itemValuePath === undefined || itemLabelPath === undefined) {
```
Comment (web-padawan) [inline src/vaadin-combo-box-mixin.html]: Same as above: please revert the change to this line in case if new logic does not require it.

File: src/vaadin-combo-box-mixin.html [hunk 6]
```diff
@@ -759,7 +771,7 @@
     }
 
     _filteredItemsChanged(e, itemValuePath, itemLabelPath) {
-      if (e.value === undefined || itemValuePath === undefined || itemLabelPath === undefined) {
```
Comment (web-padawan) [inline src/vaadin-combo-box-mixin.html]: Same as above: please revert the change to this line in case if new logic does not require it.

File: src/vaadin-combo-box-mixin.html [hunk 7]
```diff
@@ -623,10 +624,21 @@
...
+        this._inputElementValue = this._getItemLabel(this.selectedItem);
+      }
+    }
+
     _filterChanged(filter, itemValuePath, itemLabelPath) {
-      if (filter === undefined || itemValuePath === undefined || itemLabelPath === undefined) {
+      if (filter === undefined) {
         return;
       }
+
```
Comment (web-padawan) [inline src/vaadin-combo-box-mixin.html]: nit: please remove newline, so the diff would be cleaner.

File: src/vaadin-combo-box-mixin.html (hunk 5 above)
Comment (web-padawan) [reply src/vaadin-combo-box-mixin.html]: Discussed in person: new test is failing without this change. OK to include it.

File: src/vaadin-combo-box-mixin.html (hunk 6 above)
Comment (web-padawan) [reply src/vaadin-combo-box-mixin.html]: Discussed in person: new test is failing without this change. OK to include it.

File: src/vaadin-combo-box-mixin.html (hunk 4 above)
Comment (web-padawan) [reply src/vaadin-combo-box-mixin.html]: Discussed in person, OK to include this change.

File: src/vaadin-combo-box-mixin.html [hunk 8]
```diff
@@ -623,10 +624,21 @@
       }
     }
 
+    _itemLabelPathChanged(itemLabelPath, oldItemLabelPath) {
+      if (typeof itemLabelPath !== 'string') {
+        console.error('You should set itemLabelPath to a valid string');
+      }
+
+      if (this.selectedItem) {
+        this._inputElementValue = this._getItemLabel(this.selectedItem);
```
Comment (tomivirkki) [inline src/vaadin-combo-box-mixin.html]: This logic also already exists [here](https://github.com/vaadin/vaadin-combo-box/blob/master/src/vaadin-combo-box-mixin.html#L692) and it has some additional conditions to it also. Would be cleaner to reuse that logic instead of rewriting something similar here. So the fix would be to change the observer params to `'_selectedItemChanged(selectedItem, itemLabelPath)'`

I also noticed that the following code in the test suite produces different results. Can you check it's not a regression from this PR:

```
combobox.value = null;
combobox.itemLabelPath = 'custom';

console.log(combobox.inputElement.value); // "empty-string-custom"
```

```
combobox.itemLabelPath = 'custom';
combobox.value = null;

console.log(combobox.inputElement.value); // ""
```

Comment (niiyeboah) [reply src/vaadin-combo-box-mixin.html]: I have confirmed that this behaviour was there before my changes. If you change the `itemLabelPath` after the `value` change then because the `value` is set to empty string [here](https://github.com/vaadin/vaadin-combo-box/blob/adfe7b783dab90079f92a9ec283cc2427badf498/src/vaadin-combo-box-mixin.html#L679), the input element changes to the empty string value in the items, but if you change the `value` to null after the `itemLabelPath` change then the input element value is set to the empty string value [here](https://github.com/vaadin/vaadin-combo-box/blob/adfe7b783dab90079f92a9ec283cc2427badf498/src/vaadin-combo-box-mixin.html#L683).

Comment (niiyeboah) [reply src/vaadin-combo-box-mixin.html]: The `_selectedItemChanged(selectedItem, itemLabelPath)` change is done.
